crates/core-manager/src/manager.rs

```diff
@@ -0,0 +1,499 @@
+                        .map_err(|err| LoadingError::CreateCoreManager { err })?;
+                core_manager
+                    .persist()
+                    .map_err(|err| LoadingError::PersistError { err })?;
+                Ok((core_manager, task))
+            }
+        } else {
+            tracing::debug!(target: "core-manager", "The previous state was not found. Skipping...");
+            let (core_manager, task) = Self::new(state_file.clone(), system_cpu_count, cpus_range)
+                .map_err(|err| LoadingError::CreateCoreManager { err })?;
+            core_manager
+                .persist()
+                .map_err(|err| LoadingError::PersistError { err })?;
+            Ok((core_manager, task))
+        }
+    }
+
+    fn get_range(cpus_range: Option<CoreRange>) -> RangeSetBlaze<usize> {
+        cpus_range
+            .map(|range| range.0)
+            .unwrap_or(RangeSetBlaze::from_iter(0..num_cpus::get_physical()))
+    }
+}
+
+pub struct PersistenceTask {
+    receiver: Receiver<()>,
+}
+
+impl PersistenceTask {
+    async fn process_events(core_manager: Arc<CoreManager>, mut receiver: Receiver<()>) {
+        let core_manager = core_manager.clone();
+        let _ = receiver.recv().await;
+        tokio::task::spawn_blocking(move || {
+            let result = core_manager.persist();
+            match result {
+                Ok(_) => {
+                    tracing::debug!(target: "core-manager", "Core state was persisted");
+                }
+                Err(err) => {
+                    tracing::warn!(target: "core-manager", "Failed to save core state {err}");
+                }
+            }
+        })
+        .await
+        .expect("Could not join persist task")
+    }
+
+    async fn task(self, core_manager: Arc<CoreManager>) {
+        let persist_task = Self::process_events(core_manager, self.receiver);
+        tokio::pin!(persist_task);
+        loop {
+            tokio::select! {
+            _ = &mut persist_task => {
+                }
+            }
+        }
+    }
+    pub async fn run(self, core_manager: Arc<CoreManager>) {
+        tokio::task::Builder::new()
+            .name("core-manager-persist")
+            .spawn(self.task(core_manager))
+            .expect("Could not spawn persist task");
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+struct InnerState {
+    // mapping between physical and logical cores
+    cores_mapping: MultiMap<PhysicalCoreId, LogicalCoreId>,
+    // allocated system cores
+    system_cores: BTreeSet<PhysicalCoreId>,
+    // free physical cores
+    available_cores: BTreeSet<PhysicalCoreId>,
+    // mapping between physical core id and unit id
+    unit_id_mapping: BiMap<PhysicalCoreId, UnitId>,
+    // mapping between unit id and workload type
+    type_mapping: Map<UnitId, WorkerType>,
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub enum WorkerType {
+    CC,
+    Worker,
+}
```

Comment:
It's not really a worker type, right? Workers exist only in Deals.

So it probably should be `WorkType` or `Occupation` or something like that


```suggestion
pub enum WorkType {
    CapacityCommitment,
    Deal,
}
```

Reply:
Fixed